contentcuration/contentcuration/static/js/edit_channel/channel_set/views/ChannelSetList.vue

```diff
@@ -0,0 +1,266 @@
+        maxlength="200"
+    />
+
+    <h4>
+      <i class="pull-right descriptionCounter" :class="{redText: !charsLeft}">{{ $tr('charCount', {'charCount': charsLeft}) }}</i>
+      {{ $tr('descriptionLabel') }}
+    </h4>
+    <textarea
+      class="set-input"
+      v-model="description"
+      dir="auto"
+      :maxlength="charLimit"
+      rows="4"
+      :placeholder="$tr('descriptionPlaceholder')"
+    >
+    </textarea>
+    <hr/>
+
+    <!-- Channel list section -->
+    <div class="channel-list">
+      <div v-if="loadChannels" class="default-item">
+        {{ $tr('loading') }}
+      </div>
+      <div v-else>
+        <div class="channelSetMetadata">
+          <div class="pull-right" v-if="token">
+            <input
+              type="text"
+              class="copyTokenText text-center"
+              v-model="token"
+              readonly
+              ref="tokenText"
+              size='15'>
+            </input>
+            <i
+              class="material-icons copyTokenButton"
+              :title="$tr('copyTokenButtonLabel')"
+              @click="copyToken">
+              {{copyIcon}}
+            </i>
+          </div>
+          <span class="channelCountText">{{ $tr('channelCountText', {'channelCount': channelCount}) }}</span>
+        </div>
+        <div class="container-fluid">
+          <ChannelItem
+            v-for="channel in channels"
+            :key="channel.id"
+            :channel="channel"
+          />
+        </div>
+      </div>
+    </div>
+    <div id="selectChannelsButton">
+      <button
+        class="action-button uppercase"
+        @click="goToSelectChannels"
+      >
+        <i class="material-icons">add</i>
+        {{ $tr('selectButtonLabel') }}
+      </button>
+    </div>
+
+  </div>
+
+</template>
+
+
+<script>
+
+import { mapGetters, mapActions } from 'vuex';
+import ChannelItem from './ChannelItem.vue';
+
+const copyStatusCodes = {
+  IDLE: "IDLE",
+  SUCCESS: "SUCCESS",
+  FAILED: "FAILED",
+}
+
+export default {
+  name: 'ChannelSetList',
+  $trs: {
+    'loading': "Loading...",
+    'titleLabel': "Title",
+    'titlePlaceholder': "Title your collection",
+    'descriptionLabel': "Description",
+    'descriptionPlaceholder': "Describe your collection",
+    'selectButtonLabel': 'Select',
+    'channelCountText': '{channelCount, plural, =1 {# channel in your collection} other {# channels in your collection}}',
+    'copyTokenButtonLabel': "Copy Token",
+    'titleRequiredText': "Title is required",
+    "charCount": "{charCount, plural, =1 {# character left} other {# characters left}}"
+  },
+  mounted() {
+    this.loadChannelSetChannels();
+  },
+  components: {
+    ChannelItem,
+  },
+  data() {
+    return {
+      copyStatus: copyStatusCodes.IDLE,
+    }
+  },
+  computed: Object.assign(
+    mapGetters('channel_set', [
+      'channels',
+      'channelSet',
+      'isValid',
+      'loadChannels'
+    ]),
+    {
+      name: {
```

Comment:
We don't usually use the `set`/`get` pattern in kolibri because it's usually more straightforward to gave a getter for state reads and a method dedicated for mutations. I think this might be more opinion than anything, though.

Reply:
It sounds like this approach might have some additional benefits based on [this page](https://vuex.vuejs.org/guide/forms.html), but are there other benefits using the other method?

Reply:
Updated- might be good to revisit later